libs/dps/dtm0_lib.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+"""
+Library Methods for DTM0 testing
+"""
+import logging
+import random
+import time
+
+from config import S3_CFG
+from libs.s3 import ACCESS_KEY, SECRET_KEY
+from scripts.s3_bench import s3bench
+
+
+class DTM0TestLib:
+
+    def __init__(cls, access_key=ACCESS_KEY, secret_key=SECRET_KEY):
+        """
+        Init method
+        :param access_key: Access key for S3bench operations.
+        :param secret_key: Secret key for S3bench operations.
+        """
+        cls.log = logging.getLogger(__name__)
+        cls.access_key = access_key
+        cls.secret_key = secret_key
+
+    def perform_write_op(self, bucket_prefix, object_prefix, no_of_clients, no_of_samples, obj_size,
+                         log_file_prefix, queue, loop=1):
+        """
+        Perform Write operations
+        :param bucket_prefix: Bucket name
+        :param object_name: Object name
+        :param clients: No of Client session
+        :param samples: No of samples
+        :param size: Object size
+        :param log_file_prefix: Log file prefix
+        :param queue: Multiprocessing Queue to be used for returning values (Boolean,dict)
+        :param loop: Loop count for writes
+        """
+        results = list()
+        workload = list()
+        log_path = None
+        for iter_cnt in range(loop):
+            self.log.info("Iteration count: %s",iter_cnt)
+            self.log.info("Perform Write Operations : ")
+            bucket_name = bucket_prefix + str(int(time.time()))
+            resp = s3bench.s3bench(self.access_key,
```

Comment:
are we checking anywhere is s3bench is installed or not or do we want to install it as part of init maybe?

Reply:
Added in setup class as we don't want to assert in init.